service/s3/s3manager/batch.go

```diff
@@ -162,7 +161,7 @@ func (iter *DeleteListIterator) Next() bool {
 	}
 
 	if len(iter.objects) == 0 && iter.Paginator.Next() {
-		iter.objects = iter.Paginator.Page().(s3.ListObjectsOutput).Contents
+		iter.objects = iter.Paginator.CurrentPage().(s3.ListObjectsOutput).Contents
```

Comment:
The s3/s3manager/DeleteBatch can be updated to use the Pager created by S3.ListObjectsRequest.Paginate() method. 

This would also get rid of the closure here. If there are other places that have the aws.Pager{NewRequest:...} like this they probably can be updated as well.

```go
func NewDeleteListIterator(...
    pager := svc.ListObjectsRequest(input).Paginate()
    iter := &DeleteListIterator{
        Bucket: input.Bucket,
        Paginator: pager.Pager,
    }

//...
}